yammbs/_mdanalysis.py

```diff
@@ -0,0 +1,91 @@
+import MDAnalysis
+from openff.toolkit import Molecule
+
+
+def from_openff(molecule: Molecule) -> MDAnalysis.Universe:
+    return from_openff_mapped(
+        molecule,
+        molecule.to_smiles(mapped=True),
+    )
+
+
+# TODO: Caching might be useful, but needs to account for two molecules having
+#       identical (mapped) SMILES but (crucially!) different conformers.
+# @functools.lru_cache
+def from_openff_mapped(molecule: Molecule, mapped_smiles: str) -> MDAnalysis.Universe:
+    return MDAnalysis.Universe(molecule.to_rdkit())
+
+
+def get_bonds(molecule: Molecule) -> dict[tuple[int, int], float]:
+    assert molecule.n_conformers == 1, "Need a conformer for coordinate analysis"
+
+    universe = from_openff(molecule)
+
+    result = dict()
+
+    for bond in molecule.bonds:
+        atom_indices = [bond.atom1_index, bond.atom2_index]
+        bond_coords = universe.coord.positions[atom_indices]
+        # Process angle using MDAnalysis
+        result[tuple(atom_indices)] = MDAnalysis.lib.distances.calc_bonds(*bond_coords)
+
+    return result
```

Comment:
FYI, calc_bonds takes arrays of positions too. So it may be faster to do something like the below (untested):

```
positions = universe.coord.positions
atom_1s = [bond.atom1_index for bond in molecule.bonds]
atom_2s = [bond.atom2_index for bond in molecule.bonds]
bond_lengths = MDAnalysis.lib.distances.calc_bonds(positions[atom_1s], positions[atom_2s])
result = {
  (i, j): dist
  for i, j, dist in zip(atom_1s, atom_2s, bond_lengths)
}
```

Same generally applies for angles etc.